### PR TITLE
Create XDP BVT test job

### PR DESCRIPTION
**Pull request by maolson-msft**
Creates a test job for XDP which downloads the XDP build, installs duonic,
and invokes test.ps1/run-gtest.ps1 with a new "DuoNic" parameter. The tests
have not been modified to use duonic yet, so for the time being the test
is short-circuited in the beginning of run-gtest.ps1.

### CODE REVIEW

File: .azure/templates/run-bvt.yml
Comment (nibanks) [inline .azure/templates/run-bvt.yml:47]: These are not "if/else" type statements, so since the one right after will also be true, I expect you to get errors.

Comment (maolson-msft) [reply .azure/templates/run-bvt.yml:47]: I didn't get errors, but I do see in the log that it ended up with
'c:\a\w\1\s\scripts\prepare-machine.ps1' -Configuration Test -TestCertificates

so the third case must have overridden my new one.

**Review by nibanks — APPROVED** [review]

**Comment by maolson-msft** [comment]
@nibanks Don't merge this, there are still issues. I will merge when it's ready.

Currently trying to understand why the "download build artifacts" stage is skipping the xdp (and all other) binaries:
2022-03-09T23:47:49.9879185Z Skipped processing item artifacts/bin/windows/x64_Debug_schannel_xdp/msquic.dll
etc

**Comment by maolson-msft** [comment]
I've fixed the download-build-artifacts step by capitalizing "Xdp" in ExtraArtifactDir in the build job (lucky guess; turns out it's case-sensitive).

But now I notice that prepare-machine still isn't getting the -DuoNic flag:
Formatted command: . 'c:\a\w\1\s\scripts\prepare-machine.ps1' -Configuration Test -TestCertificates

I think the problem is my new case statements in run-bvt.yml checking for eq(parameters.extraArtifactsDir, 'Xdp') when they should instead be checking for '_Xdp'. Giving that a try.

**Comment by maolson-msft** [comment]
Damn, I still don't see "-DuoNic" passed to prepare-machine.ps1:

https://dev.azure.com/ms/msquic/_build/results?buildId=285638&view=logs&j=52ccb4df-c2d8-5843-0ed7-5083546684c6&t=27da6c98-2ef2-5a92-2efd-ab889fff395d&l=10

I'm stumped.

**Comment by nibanks** [comment]
> Damn, I still don't see "-DuoNic" passed to prepare-machine.ps1:
> 
> https://dev.azure.com/ms/msquic/_build/results?buildId=285638&view=logs&j=52ccb4df-c2d8-5843-0ed7-5083546684c6&t=27da6c98-2ef2-5a92-2efd-ab889fff395d&l=10
> 
> I'm stumped.

I _think_ the problem is you're passing 3 args to `and(...)` and it might ignore anything after the second arg.

**Comment by thhous-msft** [comment]
Replied with why. Theres a spelling error in the parameters.

File: .azure/templates/run-bvt.yml
Comment (maolson-msft) [reply .azure/templates/run-bvt.yml:47]: Oh dear lord :)

**Comment by maolson-msft** [comment]
> Replied with why. Theres a spelling error in the parameters.

Yep, that fixed it, great catch!

Everything looks good to me now in the logs; if somebody can approve the latest iteration I'll merge it.

**Review by nibanks — APPROVED** [review]

**Pull request merged**
